### ISSUE TITLE
ActionSheetIOS should provide a way to add accessibility to it's buttons

### ISSUE THREAD

**Issue by kevinrodriguez-io** [facebook/react-native#29401]
## Description
The ActionSheetIOS component doesn't expose any api for us to be able to customize what's being read by screen readers. This is specially hard when it comes to the role of the elements, it should at least be a `button`.

## React Native version:
Run `react-native info` in your terminal and copy the results here.

```
System:
    OS: macOS 10.15.5
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 1.25 GB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.20.1 - ~/Projects/@boltsource/react-native/node_modules/.bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.6, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 11.6/11E708 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: ^0.60.6 => 0.60.6 
  npmGlobalPackages:
    react-native-cli: 2.0.1
```

## Steps To Reproduce

1. Create a dummy app that displays an ActionSheet (The example on expo at the [docs](https://reactnative.dev/docs/actionsheetios) works), and add a couple of items.
2. Turn on VoiceOver on iOS
3. Return to the app and tap an item, for example, if I added: [ "Add Carrot To Cart", "Add Plastic To Cart" ], it should read "Add Carrot To Cart **Button**, Double Tap To Activate, instead of the current "Add Carrot To Cart, Double Tap To Activate".

## Expected Results
- it should read "TEXT_HERE **Button**, Double Tap To Activate" instead of "TEXT_HERE **Button**, Double Tap To Activate" or provide an API to customize the accessibility of the elements being used.
- Please note the **button** Speech, comes from the element accessibilityRole, which is missing.

## Snack, code example, screenshot, or link to a repository:
You can use the example snack for ActionSheetIOS on any recent version.
https://reactnative.dev/docs/actionsheetios

**Comment by kevinrodriguez-io** [comment]
Can someone add this to backlog? Of course it's going to be stale since it's accessibility, but that doesn't make it less important. A blind person might not know what is touching because of the missing role.

**Comment by blavalla** [comment]
I think this would be a great API to add to RN, but it would be very difficult to support since iOS's own UIAlertAction (which are used to render ActionSheet actions) does not support accessibility labels or other accessibility properties of any kind.

The only way around this would be to use some non-public APIs to potentially manually set these values, but there is no guarantee that these would continue to work properly as iOS is updated.

So unfortunately, I think this issue should be one that is raised with Apple, as it's an inherent limitation of the way they have designed their alerts to work.
